### PR TITLE
optimization for tx2

### PR DESCRIPTION
**Pull request by Eclipsehelio**
optimization for tx2

### CODE REVIEW

**Comment by aboarya** [comment]
@Eclipsehelio  This is very great!  I have been working on this from time to time as well for the last three weeks.  Thank you very much for this!

I think I understood your documentation.  Have you tried this on apollo 3.5 or not yet? It should almost be the same since we need to build ros on TX2 for 3.5 as well.  

Which version of Jetpack did you flash the TX2 with? Did you run Ubuntu 16.04 on TX2 and Ubuntu 14.04 docker container?

If so, did this cause any issues between the Nvidia drivers for Jetpack 3.x working in docker for Ubuntu 14.04 since they use different kernel versions?

Finally, my email is sameh4@gmx.com .. please send me a message because I would like to get in touch with you about this.

File: WORKSPACE.in [hunk 1]
```diff
@@ -94,6 +94,11 @@ new_http_archive(
     strip_prefix = "opencv-2.4.13.2",
     url = "file:///home/tmp/opencv-2.4.13.2.zip",
 )
+#new_local_repository(
```
Comment (ycool) [inline WORKSPACE.in]: this change will fail to compile //modules/tools/image_decompress, etc.

File: WORKSPACE.in [hunk 2]
```diff
@@ -142,6 +147,11 @@ new_http_archive(
     strip_prefix = "yaml-cpp-yaml-cpp-0.5.3",
     url = "file:///home/tmp/yaml-cpp-0.5.3.zip",
 )
+#new_local_repository(
```
Comment (ycool) [inline WORKSPACE.in]: please verify ./apollo.sh build in x86_64 platform too.

File: WORKSPACE.in (hunk 1 above)
Comment (Eclipsehelio) [reply WORKSPACE.in]: I didn't found target "//modules/tools/image_decompress" in "v3.1_dev". Then I checked it in "master" and found it. This PR will merge to "v3.1_dev". Finally, changes in WORKSPACE.in are commented out, from bazel's point of view this file is the same as before.

File: WORKSPACE.in (hunk 2 above)
Comment (Eclipsehelio) [reply WORKSPACE.in]: I have verified on x86_64 and ARM64 platform with these commands:
./apollo.sh build
./apollo.sh build_gpu
./apollo.sh build_opt_gpu
All Build passed!

**Pull request merged**